### PR TITLE
Character component updates

### PR DESCRIPTION
**Pull request by EmosewaMC**
Improve character component documentation as well as its component dependencies

### CODE REVIEW

File: docs/components.rst [hunk 1]
```diff
@@ -123,8 +126,6 @@ the internal name of a component class would have been :file:`LWO{Name}Component
    components/102-commendation-vendor
    components/104-rail-activator
    components/105-roller
-   components/106-unknown
-   components/107-possessor
```
Comment (Xiphoseer) [inline docs/components.rst:127]: Do we have new info on 107? Because I think it appears in the ComponentsRegistry somewhere

**Review by Xiphoseer — COMMENTED** [review]
LGTM. Is there a specific reason
you merged all XML sections into one?

File: docs/components.rst (hunk 1 above)
Comment (aronwk-aaron) [reply docs/components.rst:127]: https://lcdruniverse.org/lu_packets/lu_packets/raknet/client/replica/bbb/struct.BbbProtocol.html
107 is BBB Protocol

**Comment by EmosewaMC** [comment]
To me, having all the xml information for the character in one spot makes much more sense since everything needs to be under a specific parent node.  Could you elaborate on a way to split this into multiple files if you know what you'd like them to look like?

File: docs/components.rst (hunk 1 above)
Comment (EmosewaMC) [reply docs/components.rst:127]: Good catch. I'll add 107 in since I technically removed it.  Same with 106

**Comment by EmosewaMC** [comment]
Perhaps there could be 1 doc with just the full xml definition and then in each component I can put the sub items that would relate to that component?

**Comment by Xiphoseer** [comment]
> To me, having all the xml information for the character in one spot makes much more sense since everything needs to be under a specific parent node.  Could you elaborate on a way to split this into multiple files if you know what you'd like them to look like?

The current headings were nested as to "specific parent nodes", so the idea was to have the element structure in TOC and the attributes as list. That seems to not have been obvious at least.

I feel like if we have a single doc, it should use a standard format like xml-schema or DTD. I tend to prefer the LU packets style of having definitions with links to immediate children, but either way works.

**Comment by EmosewaMC** [comment]
i like the idea of splitting this out more as well - ill work on that

**Comment by EmosewaMC** [comment]
Is this more what you had in mind Xipho with `b24aa07` as far as formatting goes

File: docs/components.rst (hunk 1 above)
Comment (EmosewaMC) [reply docs/components.rst:127]: Added back 107 and no need to add back 106 since it was just renamed :P

**Review by Xiphoseer — APPROVED** [review]
Yeah, sure that seems prefectly reasonable to me

**Comment by Xiphoseer** [comment]
> Is this more what you had in mind Xipho with `b24aa07` as far as formatting goes

To me, attributes and child nodes are somewhat orthogonal *in XML*, because the former is structural and the latter isn't, but LU doesn't really do that here (at least not nearly as far as sth like LXFML) so it's fine.

What a DTD would add is specify how much of an element you can have where, but that doesn't really matter here either because everything is only used once, apart from repetition. And be machine transpilable to a parser or sth.

**Comment by EmosewaMC** [comment]
interesting!  It may be worth considering exploring those paths to see what looks best and documents the structure best.  After all, my main reasoning for not using headers was because indentation is very clear to me as to where children start whereas headers didn't seem to be at first glance.  Now that's just how i interpreted it having not documented XML before so my point of view could be one of an inexperienced one.

**Comment by EmosewaMC** [comment]
I just did a quick look through of DTD and i see why we could lean towards using that for a single document that has all of the XML - very cool stuff to ponder

**Pull request merged**